weaviate/backup/backup.py

```diff
@@ -0,0 +1,351 @@
+            exclude_classes: Union[List[str], str, None]=None,
+            wait_for_completion: bool=False,
+        ) -> dict:
+        """
+        Restore a backup of all/per class Weaviate objects.
+
+        Parameters
+        ----------
+        backup_id : str
+            The identifier name of the backup.
+            NOTE: Case insensitive.
+        storage_name : str
+            The storage from where to restore the backup. Currently available options are:
+                "filesystem", "s3" and "gsc".
+            NOTE: Case insensitive.
+        include_classes : Union[List[str], str, None], optional
+            The class/list of classes to be included in the backup restore. If not specified all
+            classes will be included (that were backup-ed). Either `include_classes` or
+            `exclude_classes` can be set. By default None.
+        exclude_classes : Union[List[str], str, None], optional
+            The class/list of classes to be excluded in the backup restore.
+            Either `include_classes` or `exclude_classes` can be set. By default None.
+        wait_for_completion : bool, optional
+            Whether to wait until the backup restore is done.
+
+        Returns
+        -------
+        dict
+            Backup restore response.
+
+        Raises
+        ------
+        requests.ConnectionError
+            If the network connection to weaviate fails.
+        weaviate.UnexpectedStatusCodeException
+            If weaviate reports a none OK status.
+        """
+
+        if not isinstance(backup_id, str):
+            raise TypeError(
+                f"'backup_id' must be of type str. Given type: {type(backup_id)}."
+            )
+        if storage_name not in STORAGE_NAMES:
+            raise ValueError(
+                f"'storage_name' must have one of these values: {STORAGE_NAMES}. "
+                f"Given value: {storage_name}."
+            )
+        if include_classes:
+            if isinstance(include_classes, str):
+                include_classes = [include_classes]
+            elif not isinstance(include_classes, list):
+                raise TypeError(
+                    "'include_classes' must be of type str, list of str or None. "
+                    f"Given type: {type(include_classes)}."
+                )
+        else:
+            include_classes = []
+
+        if exclude_classes:
+            if isinstance(exclude_classes, str):
+                exclude_classes = [exclude_classes]
+            elif not isinstance(exclude_classes, list):
+                raise TypeError(
+                    "'exclude_classes' must be of type str, list of str or None. "
+                    f"Given type: {type(exclude_classes)}."
+                )
+        else:
+            exclude_classes = []
+
+        if include_classes and exclude_classes:
+            raise TypeError(
+                "Either 'include_classes' OR 'exclude_classes' can be set, not both."
+            )
```

Comment:
Move this validation into an extra function? It is used at different places.

 Maybe for the future: I'm a big fan of using a dataclass as config and have the validation in there